Add methods for peak and volume of blackout hours

louispt1 opened the pull request:
Goes with https://github.com/quintel/merit/pull/157

This is a draft PR for now as I'd like to determine a test case and test the changes with the commit on Merit. Perhaps we can do that together @kaskranenburgQ?

noracato commented:
If you push you merit branch, you can specify that merit commit in the gemfile on this engine branch. That way it's easy for others to test!

louispt1 commented:
> If you push you merit branch, you can specify that merit commit in the gemfile on this engine branch. That way it's easy for others to test!

Done! Good idea

kaskranenburgQ commented:
We can test this by using the export "Electricity load curves". 
In this export the deficit for each hour is given.
We can double check if all the input and output curves in the export add up with the deficits. 

I have added ETSource queries for the table in [this commit](https://github.com/quintel/etsource/commit/bdb59462e221aeddc58cb22dd18ece32dc11f410) on [this branch](https://github.com/quintel/etsource/tree/blackout). We can test the work with these queries.

kaskranenburgQ commented:
NTH: The engine methods are now named after the old term "blackout_hours", since we no longer use this term we should rename these to power_shortage_hours.

louispt1 commented:
FYI - I have updated the names of the methods as per the NTH above. Notifying @kaskranenburgQ

mabijkerk commented on Gemfile:
```diff
@@ -74,7 +74,7 @@ gem 'parallel'
 
 # own gems
-gem 'quintel_merit', ref: '7b7a3cf', github: 'quintel/merit'
+gem 'quintel_merit', ref: 'bbce9b01788d0b696dd831dac3d1e5f675da5141', github: 'quintel/merit'
```
Why use the full hash here?

mabijkerk commented on app/models/qernel/graph_api/energy.rb:
```diff
@@ -101,10 +101,24 @@ def total_number_of_excess_events(excludes = [])
         graph.plugin(:merit).order.blackout.number_of_hours
       end
 
+      # Public: Returns the peak of blackout hours, defined as the largest single hour deficit
```
Following our discussion that documentation should be useful for both modellers and developers: it would be useful to add units to the comments of these functions: is this MJ, MW, MWh?

louispt1 commented:
@mabijkerk I've addressed your comments by adding the units to the comments in the energy.rb file and shortening the commit hash for the relevant merit commit. We may want to rebase/merge merit before merging this branch and double check the commit hash before merging it here.

mabijkerk commented on app/models/qernel/graph_api/energy.rb:
```diff
@@ -93,18 +95,32 @@ def group_of_excess_events(durations = [], excludes = [])
 
+      # Public: Returns the peak of blackout hours, defined as the largest single hour deficit
+      #
+      # Returns an Integer (MW)
```
@louispt1 I might miss some developer terminology, but integer means it's a whole number right? This returns a number with decimals, shouldn't be a float?

louispt1 commented:
These changes have already been merged to master
